lib/jobba/clause.rb

```diff
@@ -41,4 +45,84 @@ def to_new_set
     new_key
   end
 
+  def result_ids(offset: nil, limit: nil)
+    # If we have one key and it is sorted, we can let redis return limited IDs,
+    # so handle that case specially.
+
+    id_data =
+      if @keys.one?
+        # offset and limit may or may not be used, so have to do again below
+        get_members(key: @keys.first, offset: offset, limit: limit)
+      else
+        ids = @keys.flat_map do |key|
+          # don't do limiting here -- doesn't make sense til we collect all the members
+          get_members(key: key)[:ids]
+        end
+
+        ids.sort!
+        ids.uniq! unless @keys_contain_only_unique_ids
+
+        {ids: ids, is_limited: false}
+      end
+
+    if !offset.nil? && !limit.nil? && id_data[:is_limited] == false
+      id_data[:ids].slice(offset, limit)
+    else
+      id_data[:ids]
+    end
+  end
+
+  def get_members(key:, offset: nil, limit: nil)
+    if sorted_key?(key)
+      min = @min.nil? ? "-inf" : "(#{@min}"
+      max = @max.nil? ? "+inf" : "(#{@max}"
```

Comment:
Can you add some comments?

Don't quite understand why min and max are both open, but guess it doesn't matter (for now) since they seem to be used only used in timestamps?